tests/RobotFramework/tests/mqtt/disconnect_from_broker.robot

```diff
@@ -0,0 +1,71 @@
+*** Settings ***
+Resource        ../../resources/common.resource
+Library         ThinEdgeIO
+Library         OperatingSystem
+
+Suite Setup     Custom Setup
+Suite Teardown  Get Logs
+
+Test Tags      theme:mqtt
+
+
+*** Variables ***
+
+${TOPIC}             test/topic
+${MESSAGE}           Hello
+
+
+*** Test Cases ***
+
+Validate no Error message in journalctl
+    [Documentation]    This Test step is inspired by 'tedge mqtt cli commands do not disconnect cleanly from MQTT broker #2922' bug
+    ...                extended to check for any possible error that may appear in the journalctl
+    Set up broker with server and client authentication
+    Restart Service    mosquitto
+    tedge configure MQTT server authentication
+    tedge configure MQTT client authentication
+    Execute Command    tedge mqtt pub foo bar
+    ${log}    Execute Command    journalctl -u mosquitto -n 30
+    Should Not Contain    ${log}    Error
+
+Publish Connection Closed On Interruption
+    [Documentation]    Tests that a publish connection to MQTT broker closes upon interruption.
+    Set up broker with server and client authentication
+    Restart Service    mosquitto
+    tedge configure MQTT server authentication
+    tedge configure MQTT client authentication
+    ${pub_process}=    Execute Command    tedge mqtt pub ${TOPIC} ${MESSAGE} & echo $!    return_stdout=True
+    Sleep    2s    reason=Allow some time for connection
+    Kill Process    ${pub_process}
+    Process Should Not Be Running    ${pub_process}
```

Comment:
I'm pretty sure the process will never take longer than 2 seconds to publish to the MQTT broker (unless if the connection to the broker is purposfully misconfigured?)

The goal of the test should be to check if a clean MQTT disconnect is sent (can be checked by observing the log messages for error), so I would expect the assertion to include:

```
    ${log}    Execute Command    journalctl -u mosquitto -n 30
    Should Not Contain    ${log}    Error
```